helpers/GenerateResourcesAndImage.ps1

```diff
@@ -102,227 +125,284 @@ Function GenerateResourcesAndImage {
+        if ($Tags) {
+            $BuilderScriptPathInjected = $TemplatePath.Replace(".json", "-temp.json")
+            $PackerTemplateContent = Get-Content -Path $TemplatePath | ConvertFrom-Json
+            $PackerTemplateContent.builders | Add-Member -Name "azure_tags" -Value $Tags -MemberType NoteProperty
+            $PackerTemplateContent | ConvertTo-Json -Depth 3 | Out-File -Encoding Ascii $BuilderScriptPathInjected
+            $TemplatePath = $BuilderScriptPathInjected
         }
+    }
+
+    $InstallPassword = $env:UserName + [System.GUID]::NewGuid().ToString().ToUpper()
+
+    Write-Host "Validating packer template..."
+    & $PackerBinary validate `
+        "-var=client_id=fake" `
+        "-var=client_secret=fake" `
+        "-var=subscription_id=$($SubscriptionId)" `
+        "-var=tenant_id=fake" `
+        "-var=location=$($AzureLocation)" `
+        "-var=resource_group=$($ResourceGroupName)" `
+        "-var=storage_account=fake" `
+        "-var=install_password=$($InstallPassword)" `
+        "-var=allowed_inbound_ip_addresses=$($AllowedInboundIpAddresses)" `
+        "-var=azure_tags=$($TagsJson)" `
+        $TemplatePath
+    
+    if ($LastExitCode -ne 0) {
+        throw "Packer template validation failed."
+    }
 
-        if ($tags) {
-            New-AzStorageAccount -ResourceGroupName $ResourceGroupName -AccountName $storageAccountName -Location $AzureLocation -SkuName "Standard_LRS" -AllowBlobPublicAccess $AllowBlobPublicAccess -EnableHttpsTrafficOnly $EnableHttpsTrafficOnly -MinimumTlsVersion "TLS1_2" -Tag $tags
+    try {
+        # Login to Azure subscription
+        if ([string]::IsNullOrEmpty($AzureClientId)) {
+            Write-Verbose "No AzureClientId was provided, will use interactive login."
+            az login --output none
         } else {
-            New-AzStorageAccount -ResourceGroupName $ResourceGroupName -AccountName $storageAccountName -Location $AzureLocation -SkuName "Standard_LRS" -AllowBlobPublicAccess $AllowBlobPublicAccess -EnableHttpsTrafficOnly $EnableHttpsTrafficOnly -MinimumTlsVersion "TLS1_2"
+            Write-Verbose "AzureClientId was provided, will use service principal login."
+            az login --service-principal --username $AzureClientId --password $AzureClientSecret --tenant $AzureTenantId --output none
+        }
+        az account set --subscription $SubscriptionId
+        if ($LastExitCode -ne 0) {
+            throw "Failed to login to Azure subscription '$SubscriptionId'."
         }
 
-        if ([string]::IsNullOrEmpty($AzureClientId)) {
-            # Interactive authentication: A service principal is created during runtime.
-            $spDisplayName = [System.GUID]::NewGuid().ToString().ToUpper()
-            $startDate = Get-Date
-            $endDate = $startDate.AddYears(1)
+        # Check resource group
+        $ResourceGroupExists = [System.Convert]::ToBoolean((az group exists --name 'vpolikarpov-imagegen-test'));
```

Comment:
Do we need to check that exactly 'vpolikarpov-imagegen-test' RG exists?

Reply:
Good point. Thanks. Fixed.